src/main/java/org/json/simple/parser/Yylex.java

```diff
@@ -602,11 +604,11 @@ else if (zzAtEOF) {
           { sb.append('\\');
           }
         case 32: break;
-        case 21: 
-          { Double val=Double.valueOf(yytext()); return new Yytoken(Yytoken.TYPE_VALUE, val);
+        case 21:
+          { BigDecimal val = new BigDecimal(yytext()); return new Yytoken(Yytoken.TYPE_VALUE, val);
           }
-        case 33: break;
-        case 1: 
+        case 33:

```

Comment:
What is the reason for deleting the `break` statement? In this case the execution will continue with _case 1_ but it's not expected.